templates/openwisp2/settings.py

```diff
@@ -173,6 +174,47 @@
 USE_L10N = True
 USE_TZ = True
 
+#Initialize django-cors-headers
+{% if cors.enabled %}
+INSTALLED_APPS.insert(0,'corsheaders',)
+MIDDLEWARE_CLASSES.insert(0,'corsheaders.middleware.CorsMiddleware',)
+{% endif %}
+
+# CORS Whitelist
+{% if cors.whitelist_all %}                                                            # If all hosts are allowed
+CORS_ORIGIN_ALLOW_ALL = True     
+{% elif cors.regex %}                                                          # If regex is enabled
+CORS_ORIGIN_REGEX_WHITELIST = tuple({{ cors.regex_whitelist }})  
+{% else %}
+CORS_ORIGIN_WHITELIST = tuple({{ cors.whitelist }})    
+{% endif %}
+
+# CORS Allow Methods
+{% if cors.methods_defaults %}                                                                                # If default methods of corsheaders are to be added as well.
+CORS_ALLOW_METHODS = default_methods + tuple({{ cors.methods }}) 
+{% else %}                                                                                                                                # If default methods of corsheaders aren't to be added. 
+CORS_ALLOW_METHODS = {{ cors.methods }}
+{% endif %}
+
+#CORS Allow Headers
+{% if cors.headers_defaults %}                                                                                # If default headers of corsheaders are to be added as well.
```

Comment:
btw why comments in the code are so far on the right? can you move them closer to the text?

Reply:
I will. It's in my aesthetics to move the comments a bit away for neatness. But since you asked, going to align everything sidewards.

Reply:
They were misaligned due to edits. Aligning them back now.